### PR TITLE
Set batch size to 800 to resolve sqlite issues

### PR DESCRIPTION
**Pull request by dralley**
closes #3499
https://pulp.plan.io/issues/3499

### CODE REVIEW

File: plugin/pulpcore/plugin/changeset/iterator.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class BatchIterator(Iterable):
     """
 
     # Default batch size.
-    BATCH = 1000
+    BATCH = 800
```
Comment (jortel) [inline plugin/pulpcore/plugin/changeset/iterator.py]: The BatchIterator is a general purpose object for which the default of 1024 seems reasonable.  The usage specific (more appropriate) size should be passed in.

File: plugin/pulpcore/plugin/changeset/iterator.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def _collated_content(self):
             dict: A dictionary of {model_class: [content,]}
                 Each content is: PendingContent.
         """
-        for batch in BatchIterator(self.content, 1024):
```
Comment (jortel) [inline plugin/pulpcore/plugin/changeset/iterator.py]: Let's make this 800 here.

File: plugin/pulpcore/plugin/changeset/iterator.py [hunk 3]
```diff
@@ -93,7 +93,8 @@ def _collated_content(self):
...
+        # higher values may hit the sqlite expression tree size limit
```
Comment (jortel) [inline plugin/pulpcore/plugin/changeset/iterator.py]: Suggest: `should be < 1000 due to sqlite limitation. ` That way if someone changes the value later the comment is still accurate and relevant.

Comment (dralley) [reply plugin/pulpcore/plugin/changeset/iterator.py]: +1

**Pull request merged**